c10/core/StorageImpl.h

```diff
@@ -248,6 +270,29 @@ struct C10_API StorageImpl : public c10::intrusive_ptr_target {
     return &pyobj_slot_;
   }
 
+  StorageExtraMeta& get_extra_meta() {
+    if (!extra_meta_) {
+      extra_meta_ = std::make_unique<StorageExtraMeta>();
+    }
+    return *extra_meta_;
+  }
+
+  [[noreturn]] void throw_data_ptr_access_error() const {
+    if (extra_meta_ && extra_meta_->custom_data_ptr_error_msg_) {
+      // NOLINTNEXTLINE(bugprone-unchecked-optional-access)
+      TORCH_CHECK(false, *extra_meta_->custom_data_ptr_error_msg_);
+    }
+    TORCH_CHECK(
+        false, "Cannot access data pointer of Storage that is invalid.");
+  }
```

Comment:
Don't put this impl in the header, move the error testing code to cpp so it doesn't get inlined, you're going to blow up code size a lot if you don't